blackboxopt/optimizers/testing.py

```diff
@@ -148,6 +150,70 @@ def is_deterministic_with_fixed_seed(
     return True
 
 
+def is_deterministic_when_reporting_shuffled_evaluations(
+    optimizer_class: Union[
+        Type[SingleObjectiveOptimizer], Type[MultiObjectiveOptimizer]
+    ],
+    optimizer_kwargs: dict,
+) -> bool:
+    """Check if determinism isn't affected by the order of initially reported data.
+
+    Repeatedly initialize the optimizer with the same parameter space and a fixed seed.
+    Report a set of initial evaluations in randomized order as initial data. Start
+    optimizing and check if the generated configurations for all optimizers are equal.
+
+    Args:
+        optimizer_class: Optimizer to test.
+        optimizer_kwargs: Expected to contain additional arguments for initializing
+            the optimizer. (`search_space` and `objective(s)` are set automatically
+            by the test.)
+
+    Returns:
+        `True` if the test is passed.
+    """
+
+    space = ps.ParameterSpace()
+    space.add(ps.ContinuousParameter("p1", [0, 1]))
+
+    def _run_experiment_1d(es):
+        x = es.configuration["p1"]
+        _x = np.copy(np.atleast_2d(x))
+        params = np.array([0.75, 0.0, -10.0, 0.0, 0.0])
+        y = np.polyval(params, _x)
+        return float(np.squeeze(y))
+
+    optimization_runs: Dict[int, list] = {0: [], 1: []}
+    for run_idx in optimization_runs:
+        opt = _initialize_optimizer(
+            optimizer_class,
+            optimizer_kwargs,
+            objective=Objective("loss", False),
+            objectives=[Objective("loss", False)],
+            space=space,
```

Comment:
Would it makes sense to set the `seed` argument here explicitly to make it obvious that the underlying rng is fixed? I know that it is fixed by default in `_initialize_optimizer`, but I actually needed to go to that function and check to convince myself that the seed is fixed.

Reply:
Good point!

Reply:
Done